src/Features/Core/Portable/Snippets/SnippetProviders/AbstractSnippetProvider.cs

```diff
@@ -107,8 +107,8 @@ public async Task<SnippetChange> GetSnippetAsync(Document document, int position
             var documentWithIndentation = await AddIndentationToDocumentAsync(reformattedDocument, position, syntaxFacts, cancellationToken).ConfigureAwait(false);
 
             var reformattedRoot = await documentWithIndentation.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
-            var caretTarget = reformattedRoot.GetAnnotatedNodes(_cursorAnnotation).FirstOrDefault();
-            var mainChangeNode = reformattedRoot.GetAnnotatedNodes(_findSnippetAnnotation).FirstOrDefault();
+            var caretTarget = reformattedRoot.GetAnnotatedNodes(_cursorAnnotation).First();
+            var mainChangeNode = reformattedRoot.GetAnnotatedNodes(_findSnippetAnnotation).First();
```

Comment:
these should not change.

Reply:
From what I can tell, these can't be null. but I will change this to `FirstOrDefault` + `ThrowIfNull`. @akhera99 maybe need to follow up on this to ensure things are in order.

Reply:
Yep, just ping me when you need me to take a look